transaction/src/auth_hash.rs

```diff
@@ -213,6 +215,57 @@ impl spend::Body {
     }
 }
 
+impl swap::Body {
+    fn auth_hash(&self) -> Hash {
+        let mut state = blake2b_simd::Params::default()
+            .personal(b"PAH:swap_body")
+            .to_state();
+
+        // All of these fields are fixed-length, so we can just throw them
+        // in the hash one after the other.
+        // TODO: actually the trading pair isn't necessarily fixed-length
+        // right now, does this have implications?
+        state.update(&self.trading_pair.encode_to_vec());
```

Comment:
The simplest thing here is probably to hash the assets in the pair in order.

Reply:
There's a `to_bytes` impl that results in a 64-byte fixed length encoding, that should be suitable? We could also add an `auth_hash` method that hashed that but I'm not sure why we'd want to.

Reply:
Ah but the `to_bytes` impl is only `pub(crate)` -- maybe it does make sense to add `auth_hash` then